### PR TITLE
Replace special chars with empty space

### PR DESCRIPTION
**Pull request by Akrion**
- Replace all special chars with ` \\u0020` to prevent undesired search behavior
- Update tests

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>We were able to automatically fix some formatting issues in this PR for you!</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>You reduced the total lines of code! Awesome! :+1:</td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: 9bdd22fdef8c58e219200c05453789c02508eb75


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 593cecfb2614b1da9a8362d6e90d7173d96114e9
</p>

File: src/example-types/filters/tagsQuery.js [hunk 1]
```diff
@@ -25,16 +25,14 @@ let addQuotesAndDistance = _.curry((tag, text) => {
...
 // https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#_reserved_characters
-let escapeReservedChars = _.flow(
+let replaceReservedChars = _.flow(
   _.toString,
-  // Escape these characters
-  _.replace(/([+\-=&|!(){}[\]^"~*?:\\/])/g, '\\$1'),
-  // Remove these characters
-  _.replace(/[><]/g, '')
+  // Replace characters with white space (\\u0020)
+  _.replace(/([+\-=&|!(){}[\]^"~*?:\\/<>;,$'])/g, '\\u0020')
```
Comment (daedalus28) [inline src/example-types/filters/tagsQuery.js:31]: `u0020` is unicode for space - we should just replace them with a space character

File: test/example-types/filters/tagsQuery.js [hunk 2]
```diff
@@ -61,18 +61,12 @@ describe('addQuotesAndDistance', () => {
...
-  it('should remove unescapable reserved characters', () => {
-    expect(escapeReservedChars('1 < 2 > 1')).to.deep.equal('1  2  1')
-  })
-  it('should not escape non-reserved characters', () => {
-    expect(escapeReservedChars(`I'm just a plain old sentence.`)).to.deep.equal(
-      `I'm just a plain old sentence.`
+    expect(
+      replaceReservedChars('foo: [bar] (baz) - 1 ^ 2 <> 3 !$ 4')
+    ).to.deep.equal(
+      'foo\\u0020 \\u0020bar\\u0020 \\u0020baz\\u0020 \\u0020 1 \\u0020 2 \\u0020\\u0020 3 \\u0020\\u0020 4'
```
Comment (daedalus28) [inline test/example-types/filters/tagsQuery.js:69]: see above - u0020 is just space so we should just use that

**Review by daedalus28 — CHANGES_REQUESTED** [review]

**Comment by Akrion** [comment]
Changes done

**Review by stellarhoof — APPROVED** [review]
What was the problem with escaping them?

**Comment by daedalus28** [comment]
> What was the problem with escaping them?

Apparently escaping an asterisk doesn't stop it from working. We have empirical evidence that a query with a ton of asterisks in the keywords triggered downtime. Either way, the analyzers on the elastic side are set up to replace these characters with spaces, so there's no point in even allowing them to be searched while escaped since they're not even in the index anyway.

**Pull request merged**